mypy/typeanal.py

```diff
@@ -952,7 +954,11 @@ def visit_callable_type(self, t: CallableType, nested: bool = True) -> Type:
                 variables = t.variables
             else:
                 variables, _ = self.bind_function_type_variables(t, t)
-            special = self.anal_type_guard(t.ret_type)
+            first_arg = t.arg_names[0] if t.arg_names else None
+            if isinstance(t.definition, FuncDef) and t.definition.info:
+                # We don't care about old typeguard on a class because it's too screwed up
```

Comment:
how so?

Reply:
Given:
``` py
class A:
    def foo(self, x) -> TypeGuard[int]: ...
```
We don't know which argument is supposed to be type guarded at the time this check is performed.

I guess I could look into moving the check to a different time.

Reply:
so does your change disable old type guards on classes? or leave the functionality as is?

Reply:
Leaves them as is